### DISCUSSION TITLE
can't install scipy 0.10.1

### DISCUSSION THREAD

**Discussion by robonut5** [neonbjb/tortoise-tts#397] [Q&A]
[New Text Document (2).txt](https://github.com/neonbjb/tortoise-tts/files/11206301/New.Text.Document.2.txt)
I'm not experienced using Python, and I don't know how to fix this. Anything you guys know that could help?

**Comment by Codylove** [comment]
So I'm no way shape or form a programmer I just learned just enough to run the programs on git, but it seems that they typo'd in the requirements, It needs to be 1.10.1 not 0.10.1. Go ahead and change it. I've also seen there being even more issues that another person brought up that you might run into that i did. its question #394. think that links it but if not 
set up a clean anaconda environment using python 3.8 threw an Admin launched Anaconda prompt
conda create --name py38 python=3.8

then every time you load into the program after install it doesn't matter if its admin launched and you wont have to create the environment every time,
First thing you do every time you launch will need to get into the environment via

conda activate py38


then get your self into the directory via cd, if its on a different drive make sure to include /d to get into the new drive -
cd /d "put the install directory inside these quotes"

for the first time installing manually install numpy, llvm, and pydantic using -

conda install -c cctbx202208 numpy
conda install -c main llvm
pip install pydantic==1.9.1

make sure you have changed the requirements.txt file to say 1.10.1

then install as instructed from there.

Hopefully this gets seen by the correct people and fixes will come for the issues.

**Reply by robonut5** [reply]
Thanks, I tried all of this, but I ran into a different error 

(py38) C:\Users\origi\tortoise-tts>python tortoise/do_tts.py --text "I'm going to speak this" --voice random --preset fast
Traceback (most recent call last):
  File "tortoise/do_tts.py", line 27, in <module>
    tts = TextToSpeech(models_dir=args.model_dir)
  File "C:\Users\origi\tortoise-tts\tortoise\api.py", line 231, in __init__
    self.autoregressive.load_state_dict(torch.load(get_model_path('autoregressive.pth', models_dir)))
  File "C:\ProgramData\anaconda3\envs\py38\lib\site-packages\torch\serialization.py", line 797, in load
    with _open_zipfile_reader(opened_file) as opened_zipfile:
  File "C:\ProgramData\anaconda3\envs\py38\lib\site-packages\torch\serialization.py", line 283, in __init__
    super().__init__(torch._C.PyTorchFileReader(name_or_buffer))
RuntimeError: PytorchStreamReader failed reading zip archive: failed finding central directory

Any idea how to fix it?

**Comment by Codylove** [comment]
Zero idea, try a fresh install from the top and look up how to destroy and recreate anaconda environments. Other wise try good searching the error code.On Apr 16, 2023, at 19:48, robonut5 ***@***.***> wrote:﻿
Thanks, I tried all of this, but I ran into a different error
(py38) C:\Users\origi\tortoise-tts>python tortoise/do_tts.py --text "I'm going to speak this" --voice random --preset fast
Traceback (most recent call last):
File "tortoise/do_tts.py", line 27, in 
tts = TextToSpeech(models_dir=args.model_dir)
File "C:\Users\origi\tortoise-tts\tortoise\api.py", line 231, in init
self.autoregressive.load_state_dict(torch.load(get_model_path('autoregressive.pth', models_dir)))
File "C:\ProgramData\anaconda3\envs\py38\lib\site-packages\torch\serialization.py", line 797, in load
with _open_zipfile_reader(opened_file) as opened_zipfile:
File "C:\ProgramData\anaconda3\envs\py38\lib\site-packages\torch\serialization.py", line 283, in init
super().init(torch._C.PyTorchFileReader(name_or_buffer))
RuntimeError: PytorchStreamReader failed reading zip archive: failed finding central directory
Any idea how to fix it?

**Comment by cuckookernel** [comment]
I believe the scipy version in requirements is an error... 0.10.1 is super-old... I changed it to 1.10.1 and the installation went through perfectly.

**Comment by sordidloam** [comment]
fixed that one, but now I have this...

`File "C:\Users\Admin\AppData\Local\Temp\pip-install-oc8jnrag\numpy_f6d1a46a5df742e9a91415143ab37971\numpy\distutils\ccompiler.py", line 90, in <lambda>
          m = lambda self, *args, **kw: func(self, *args, **kw)
      TypeError: CCompiler_spawn() got an unexpected keyword argument 'env'
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.

(env_python3.10) C:\Users\Admin>
(env_python3.10) C:\Users\Admin>conda install pytorch torchvision torchaudio pytorch-cuda=11.7 -c pytorch -c nvidia
Collecting package metadata (current_repodata.json): done
Solving environment: done`

**Reply by Byron-Zoomry** [reply]
I got the same errors as you, currently stuck on this one :(

**Comment by user76765** [comment]
So, im an idiot. but it worked.

I went into the requirments.txt and updated the scipy version to the latest one. and i installed the latest version of scipy. I also had an issue with numpy==1.20.0 - and in this case i downloaded it from conda forge
